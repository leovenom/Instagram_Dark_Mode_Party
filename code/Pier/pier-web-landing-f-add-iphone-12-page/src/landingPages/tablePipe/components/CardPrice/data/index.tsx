const data = {
  today: {
    title: "Antes da atualização",
    subtitle: (date) => `Anterior a ${date}`,
  },
  nextPrice: {
    title: "Após a atualização",
    subtitle: (date) => `A partir de ${date}`,
  },
  monthlyPayment: "Mensalidade",
  plan: "Plano",
  economic: "Econômico",
  premium: "Premium",
  totalInsuranceValue: "Valor do reembolso",
  insuranceValueUsed: "Valor de um semi-novo: ",
  samePrice: {
    imageSrc: "/static_assets/images/rebranding/woman-with-smartphone.svg",
    imageDescription:
      "Pessoa pensativa ao lado de um celular indicando queda de preço",
    title: "Sem reajustes",
    description: "Nada mudou para você nessa atualização!",
  },
};

export default data;
