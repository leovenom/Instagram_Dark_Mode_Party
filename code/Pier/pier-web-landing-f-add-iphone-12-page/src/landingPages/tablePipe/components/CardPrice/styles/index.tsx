import styled, { css } from "styled-components";

import breakpoints from "ui/theme/breakpoints";

const CardContainer = styled.div`
  box-sizing: border-box;
  max-width: 500px;
  min-height: 385px;
  width: 100%;
  padding: 32px 18px;
  background-color: #fff;

  ${breakpoints.desktop} {
    padding: 32px 25px;
  }
`;

const CardHeader = styled.div`
  margin-bottom: 36px;
  display: flex;
  align-items: center;
  flex-direction: column;
`;

const CardContent = styled.div`
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
`;

const CardRow = styled.div`
  box-sizing: border-box;
  display: flex;
  align-items: center;
  padding: 24px 0;
  border-bottom: 1px solid rgba(201, 203, 209, 0.5);

  &:first-child {
    padding-top: 0px;
  }

  &:last-of-type {
    border: none;
    margin-bottom: 30px;
  }
`;

const CardRowItem = styled.div<{ arrow?: string }>`
  width: 33%;
  display: flex;
  align-items: center;

  &:first-child {
    width: 34%;
  }

  p {
    text-overflow: ellipsis;
    overflow: hidden;
  }

  &:after {
    display: none;
    content: "";
    width: 0;
    height: 0;
    border-left: 5px solid transparent;
    border-right: 5px solid transparent;
  }

  ${({ arrow }) =>
    arrow === "down" &&
    css`
      &:after {
        margin: 0 0 3px 8px;
        display: inline-block;
        border-top: 6px solid ${({ theme }) => theme.colors.secondary};
      }
    `}

  ${({ arrow }) =>
    arrow === "up" &&
    css`
      &:after {
        margin: 0 0 3px 8px;
        display: inline-block;
        border-bottom: 6px solid ${({ theme }) => theme.colors.secondary};
      }
    `}
`;

const ImgWoman = styled.img`
  display: block;
  height: 116px;
  margin-bottom: 36px;
`;
export {
  CardContainer,
  CardHeader,
  CardContent,
  CardRow,
  CardRowItem,
  ImgWoman,
};
