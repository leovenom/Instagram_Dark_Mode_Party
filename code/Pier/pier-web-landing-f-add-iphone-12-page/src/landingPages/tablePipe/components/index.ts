import Header from "./Header";
import Instructions from "./Instructions";
import FindSmartphoneModel from "./FindSmartphoneModel";
import Faq from "./Faq";
import ShowPrices from "./ShowPrices";

export { Header, Instructions, FindSmartphoneModel, Faq, ShowPrices };
