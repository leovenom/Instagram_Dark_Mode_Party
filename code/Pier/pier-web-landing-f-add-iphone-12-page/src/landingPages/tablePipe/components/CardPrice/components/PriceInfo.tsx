import * as React from "react";

import Text from "ui/Text";
import data from "../data";

import {
  getArrowsUiOrientation,
  getMonthlyValue,
  getInsuredValue,
} from "../../utils";
import { CardRow, CardRowItem } from "../styles";

interface PriceInfo {
  smartphoneSelected: {
    plansValues: object;
    name: string;
    memory: string;
    model: {
      name: string;
    };
  };
  variant: string;
}

const PriceInfo: React.FC<PriceInfo> = ({ smartphoneSelected, variant }) => {
  const arrowsOrientation = getArrowsUiOrientation({
    smartphoneSelected,
    variant,
  });

  return (
    <>
      <CardRow>
        <CardRowItem>
          <Text variant="bodySmall" color="gray600">
            {data?.plan}
          </Text>
        </CardRowItem>
        <CardRowItem>
          <Text
            variant="bodySmallest"
            bold
            transform="uppercase"
            color="primary400"
          >
            {data?.economic}
          </Text>
        </CardRowItem>
        <CardRowItem>
          <Text
            variant="bodySmallest"
            bold
            transform="uppercase"
            color="primary400"
          >
            {data?.premium}
          </Text>
        </CardRowItem>
      </CardRow>

      <CardRow>
        <CardRowItem>
          <Text variant="bodySmall" color="gray600">
            {data?.monthlyPayment}
          </Text>
        </CardRowItem>
        <CardRowItem arrow={arrowsOrientation?.monthly?.economic}>
          <Text variant="bodySmall" color="text">
            {getMonthlyValue("economic", smartphoneSelected)?.formatted}
          </Text>
        </CardRowItem>
        <CardRowItem arrow={arrowsOrientation?.monthly?.premium}>
          <Text variant="bodySmall" color="text">
            {getMonthlyValue("premium", smartphoneSelected)?.formatted}
          </Text>
        </CardRowItem>
      </CardRow>

      <CardRow>
        <CardRowItem>
          <Text variant="bodySmall" color="gray600">
            {data?.totalInsuranceValue}
          </Text>
        </CardRowItem>
        <CardRowItem arrow={arrowsOrientation?.insuredValue?.economic}>
          <Text variant="bodySmall" color="text">
            {getInsuredValue("economic", smartphoneSelected)?.formatted}
          </Text>
        </CardRowItem>
        <CardRowItem arrow={arrowsOrientation?.insuredValue?.premium}>
          <Text variant="bodySmall" color="text">
            {getInsuredValue("premium", smartphoneSelected)?.formatted}
          </Text>
        </CardRowItem>
      </CardRow>

      <Text variant="body" align="center">
        {data.insuranceValueUsed}{" "}
        <strong>
          {getInsuredValue("premium", smartphoneSelected)?.formatted}
        </strong>
      </Text>
    </>
  );
};

export default PriceInfo;
