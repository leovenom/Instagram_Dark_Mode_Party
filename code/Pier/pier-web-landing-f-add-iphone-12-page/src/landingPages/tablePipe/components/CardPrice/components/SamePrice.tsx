import Text from "ui/Text";
import data from "../data";

import { ImgWoman } from "../styles";

const SamePrice = () => (
  <>
    <ImgWoman
      src={data.samePrice.imageSrc}
      alt={data.samePrice.imageDescription}
    />
    <Text
      variant="tag"
      bold
      transform="uppercase"
      align="center"
      color="primary400"
      mb={10}
    >
      {data.samePrice.title}
    </Text>
    <Text variant="bodySmall" align="center" color="gray600">
      {data.samePrice.description}
    </Text>
  </>
);

export default SamePrice;
