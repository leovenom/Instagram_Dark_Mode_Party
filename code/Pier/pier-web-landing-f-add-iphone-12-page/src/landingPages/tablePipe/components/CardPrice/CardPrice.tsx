import { nextPipeUpdateAt } from "helpers/pipePlansValues";
import * as React from "react";

import Text from "ui/Text";
import { PriceInfo, SamePrice } from "./components";

import data from "./data";
import { CardContainer, CardHeader, CardContent } from "./styles";

interface CardPriceProps {
  smartphoneSelected: {
    plansValues: object;
    name: string;
    memory: string;
    model: {
      name: string;
    };
  };
  variant: string;
  hasTheSamePrice?: boolean;
}

const CardPrice: React.FC<CardPriceProps> = ({
  smartphoneSelected,
  variant,
  hasTheSamePrice = false,
}) => {
  const date =
    variant === "nextPrice"
      ? nextPipeUpdateAt.format("DD/MM/YYYY")
      : "23/11/2020";
  return (
    <CardContainer>
      <CardHeader>
        <Text variant="subtitleSmall" bold color="primary" mb={8}>
          {data?.[variant]?.title}
        </Text>
        <Text variant="bodySmall" color="gray600">
          {data?.[variant]?.subtitle(date)}
        </Text>
      </CardHeader>

      <CardContent>
        {hasTheSamePrice ? (
          <SamePrice />
        ) : (
          <PriceInfo
            smartphoneSelected={smartphoneSelected}
            variant={variant}
          />
        )}
      </CardContent>
    </CardContainer>
  );
};

export default CardPrice;
