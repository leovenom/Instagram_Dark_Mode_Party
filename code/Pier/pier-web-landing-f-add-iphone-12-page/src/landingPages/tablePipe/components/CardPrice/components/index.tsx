import PriceInfo from "./PriceInfo";
import SamePrice from "./SamePrice";

export { PriceInfo, SamePrice };
