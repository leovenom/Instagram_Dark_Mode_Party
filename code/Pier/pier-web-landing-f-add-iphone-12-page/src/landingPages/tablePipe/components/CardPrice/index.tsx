import CardPrice from "./CardPrice";

export default CardPrice;
