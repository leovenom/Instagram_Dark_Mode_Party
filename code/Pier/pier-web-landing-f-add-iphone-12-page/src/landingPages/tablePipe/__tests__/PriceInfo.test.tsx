import { renderWithTheme } from "helpers/testUtils";
import maskMoney from "helpers/maskMoney";
import { PriceInfo } from "../components/CardPrice/components";

const renderComponent = ({ smartphoneSelected, variant = "today" }) =>
  renderWithTheme(
    <PriceInfo variant={variant} smartphoneSelected={smartphoneSelected} />
  );

const selectedSmartphone = {
  deviceOs: "ios",
  manufacturer: undefined,
  memory: "16GB",
  model: { memories: ["16GB", "32GB", "64GB", "128GB"], name: "iPhone 6S" },
  plansValues: {
    premium: { insuredValue: 712.0, price: 13.9 },
    economic: { insuredValue: 890.0, price: 19.1 },
  },
  serie: "Apple",
};

describe("<PriceInfo />", () => {
  it("Should render without errors", () => {
    const { container } = renderComponent({
      smartphoneSelected: selectedSmartphone,
    });
    expect(container).toMatchSnapshot();
  });

  it("Should render the component with the correct info", () => {
    const { getByText, getAllByText } = renderComponent({
      smartphoneSelected: selectedSmartphone,
    });

    // Economic Montlhy Plan and Economic Insured Value
    getByText(maskMoney(selectedSmartphone.plansValues.economic.price));
    getByText(maskMoney(selectedSmartphone.plansValues.economic.insuredValue));

    // Premium Montlhy Plan and Economic Insured Value
    getByText(maskMoney(selectedSmartphone.plansValues.premium.price));
    getAllByText(
      maskMoney(selectedSmartphone.plansValues.premium.insuredValue)
    );
  });
});
