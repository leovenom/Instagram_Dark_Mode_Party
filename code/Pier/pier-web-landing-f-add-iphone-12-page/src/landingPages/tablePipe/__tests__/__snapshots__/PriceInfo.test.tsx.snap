// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PriceInfo /> Should render without errors 1`] = `
<div>
  <div
    class="styles__CardRow-v1rd0i-3 kZOViC"
  >
    <div
      class="styles__CardRowItem-v1rd0i-4 leoyfg"
    >
      <p
        class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 kJMJxo"
        color="gray600"
      >
        Plano
      </p>
    </div>
    <div
      class="styles__CardRowItem-v1rd0i-4 leoyfg"
    >
      <p
        class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 cDGqPZ"
        color="primary400"
        transform="uppercase"
      >
        Econ√¥mico
      </p>
    </div>
    <div
      class="styles__CardRowItem-v1rd0i-4 leoyfg"
    >
      <p
        class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 cDGqPZ"
        color="primary400"
        transform="uppercase"
      >
        Premium
      </p>
    </div>
  </div>
  <div
    class="styles__CardRow-v1rd0i-3 kZOViC"
  >
    <div
      class="styles__CardRowItem-v1rd0i-4 leoyfg"
    >
      <p
        class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 kJMJxo"
        color="gray600"
      >
        Mensalidade
      </p>
    </div>
    <div
      class="styles__CardRowItem-v1rd0i-4 leoyfg"
    >
      <p
        class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 hEfGWY"
        color="text"
      >
        R$ 19,10
      </p>
    </div>
    <div
      class="styles__CardRowItem-v1rd0i-4 leoyfg"
    >
      <p
        class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 hEfGWY"
        color="text"
      >
        R$ 13,90
      </p>
    </div>
  </div>
  <div
    class="styles__CardRow-v1rd0i-3 kZOViC"
  >
    <div
      class="styles__CardRowItem-v1rd0i-4 leoyfg"
    >
      <p
        class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 kJMJxo"
        color="gray600"
      >
        Valor do reembolso
      </p>
    </div>
    <div
      class="styles__CardRowItem-v1rd0i-4 leoyfg"
    >
      <p
        class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 hEfGWY"
        color="text"
      >
        R$ 890,00
      </p>
    </div>
    <div
      class="styles__CardRowItem-v1rd0i-4 leoyfg"
    >
      <p
        class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 hEfGWY"
        color="text"
      >
        R$ 712,00
      </p>
    </div>
  </div>
  <p
    class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 difqfl"
  >
    Valor de um semi-novo: 
     
    <strong>
      R$ 712,00
    </strong>
  </p>
</div>
`;
